snuba/snuba_migrations/events_analytics_platform/0017_span_attribute_table_v3.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import annotations
+
+from typing import Sequence
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+from snuba.utils.constants import ATTRIBUTE_BUCKETS
+from snuba.utils.schemas import (
+    Column,
+    DateTime,
+    Float,
+    SimpleAggregateFunction,
+    String,
+    UInt,
+)
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    This migration creates a table meant to store just the attributes seen in a particular org.
+    The table is populated by a separate materialized view for each type of attribute.
+    """
+
+    blocking = False
+    storage_set_key = StorageSetKey.EVENTS_ANALYTICS_PLATFORM
+    granularity = "8192"
+
+    old_str_mv = "spans_str_attrs_2_mv"
+    old_str_local_table = "spans_str_attrs_2_local"
+    old_str_dist_table = "spans_str_attrs_2_dist"
+    new_str_mv = "spans_str_attrs_3_mv"
+    new_str_local_table = "spans_str_attrs_3_local"
+    new_str_dist_table = "spans_str_attrs_3_dist"
+    str_columns: Sequence[Column[Modifiers]] = [
+        Column("organization_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("attr_key", String(modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column("attr_value", String(modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column(
+            "timestamp",
+            DateTime(modifiers=Modifiers(codecs=["DoubleDelta", "ZSTD(1)"])),
```

Comment:
was doubledelta recommended against? i forget what altinity consultant said

Reply:
Not for timestamps, that's the only place where he recommended it.